File: examples/js/geocoding/lookup/geocodeAnAddress.js
const { GeocodingService } = require("m3o/geocoding");

// Lookup returns a geocoded address including normalized address and gps coordinates. All fields are optional, provide more to get more accurate results
async function geocodeAnAddress() {
  let geocodingService = new GeocodingService(process.env.MICRO_API_TOKEN);
  let rsp = await geocodingService.lookup({
    address: "10 russell st",
    city: "london",
    country: "uk",
    postcode: "wc2b",
  });
  console.log(rsp);
}

geocodeAnAddress();

File: examples/go/geocoding/lookup/geocodeAnAddress.go
package example

import (
	"fmt"
	"os"

	"github.com/micro/services/clients/go/geocoding"
)

// Lookup returns a geocoded address including normalized address and gps coordinates. All fields are optional, provide more to get more accurate results
func GeocodeAnAddress() {
	geocodingService := geocoding.NewGeocodingService(os.Getenv("MICRO_API_TOKEN"))
	rsp, err := geocodingService.Lookup(&geocoding.LookupRequest{
		Address:  "10 russell st",
		City:     "london",
		Country:  "uk",
		Postcode: "wc2b",
	})
	fmt.Println(rsp, err)
}
